src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const obj = Object.assign({}, state);
+  const result = [];
+  const actionList = [ ...actions ];
+
+  actionList.forEach(element => {
+    switch (element['type']) {
+      case 'clear':
+        for (const key in obj) {
+          delete obj[key];
+        }
+        result.push({ ...obj });
```

Comment:
move it after switch block, do it onle once

Reply:
move what?
task:
-`return an array of the same length as `actions` containing all previous versions of the `state`.